caas/kubernetes/provider/k8s.go

```diff
@@ -84,23 +90,25 @@ func (k *kubernetesClient) EnsureOperator(appName, agentPath string, newConfig c
 }
 
 func (k *kubernetesClient) ensureConfigMap(appName string, newConfig caas.NewOperatorConfigFunc) (string, error) {
-	mapName := podName(appName) + "-config"
+	mapName := configMapName(appName)
 
 	exists, err := k.configMapExists(mapName)
 	if err != nil {
 		return "", errors.Trace(err)
-	}
-	if exists {
-		logger.Infof("ConfigMap %s already exists", mapName)
-	} else {
-		config, err := newConfig(appName)
-		if err != nil {
-			return "", errors.Annotate(err, "creating config")
-		}
-		if err := k.createConfigMap(mapName, config); err != nil {
-			return "", errors.Annotate(err, "creating ConfigMap")
+	} else if exists {
+		logger.Debugf("ConfigMap %s already exists - deleting", mapName)
+		if err := k.deleteConfigMap(mapName); err != nil {
```

Comment:
can we just update the config map if it already exists? rather than making 2 API calls

Reply:
yep